src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -79,6 +82,15 @@ public FunctionsSyncManager(IConfiguration configuration, IHostIdProvider hostId
             _azureBlobStorageProvider = azureBlobStorageProvider;
         }
 
+        // Do not use in production. A method for testability.
+        internal IConfiguration Configuration
```

Comment:
I'd rather not add this. If you're trying to mock test, just pass the mock instance in on construction as needed

Reply:
We can mock the IConfiguration, however, if we do it, We can not test the behavior with`HostJsonFileConfigurationSource ` and `ScriptEnvironmentVariablesConfigurationSource`  as an integration testing. If it's ok we don't need to test the behavior, I'd switch to the mock.

Reply:
Removed